src/JuMP.jl

```diff
@@ -774,6 +774,39 @@ work for JuMP types) and exploits the mutability of `AffExpr` and `QuadExpr`.
 """
 abstract type AbstractJuMPScalar <: _MA.AbstractMutable end
 
+function _get_index_keyword_indexing_error()
+    return ErrorException(
+        """
+        Indexing an `Array` with keyword arguments is not supported.
+
+        Indexing with keyword arguments _is_ supported for `Container.DenseAxisArray`
+        (and `Container.SparseAxisArray`).
+
+        Force the container type by passing `container = DenseAxisArray` to any of
+        the JuMP macros. For example:
+
+        ```julia
+        julia> model = Model();
+
+        julia> @variable(model, x[i=1:3], container = DenseAxisArray)
+        1-dimensional DenseAxisArray{VariableRef,1,...} with index sets:
+            Dimension 1, Base.OneTo(3)
+        And data, a 3-element Vector{VariableRef}:
+        x[1]
+        x[2]
+        x[3]
+
+        julia> x[i=2]
+        x[2]
+        ```
+        """,
+    )
+end
+
+function Base.getindex(x::Array{<:AbstractJuMPScalar}; kwargs...)
```

Comment:
Isn't that called when you do `getindex(x)` without any keyword argument ? There should be at least one

Reply:
What does it mean to call `getindex(x)`?

Even without JuMP:
```Julia
julia> x = rand(3)
3-element Vector{Float64}:
 0.8542832661650948
 0.36511705612410084
 0.12183318054742132

julia> x[]
ERROR: BoundsError: attempt to access 3-element Vector{Float64} at index []
Stacktrace:
 [1] throw_boundserror(A::Vector{Float64}, I::Tuple{})
   @ Base ./abstractarray.jl:651
 [2] checkbounds
   @ ./abstractarray.jl:616 [inlined]
 [3] _getindex
   @ ./abstractarray.jl:1196 [inlined]
 [4] getindex(::Vector{Float64})
   @ Base ./abstractarray.jl:1170
 [5] top-level scope
   @ REPL[451]:1
```
But yeah, I guess we could throw a bounds error if there are no kwargs and we get dispatched here.

Reply:
Updated